Lib/posixpath.py

```diff
@@ -359,31 +337,25 @@ def normpath(path):
         path = os.fspath(path)
         if isinstance(path, bytes):
             sep = b'/'
-            empty = b''
             dot = b'.'
             dotdot = b'..'
         else:
             sep = '/'
-            empty = ''
             dot = '.'
             dotdot = '..'
-        if path == empty:
+        if not path:
             return dot
         _, initial_slashes, path = splitroot(path)
-        comps = path.split(sep)
         new_comps = []
-        for comp in comps:
-            if comp in (empty, dot):
+        for comp in path.split(sep):
+            if not comp or comp == dot:
                 continue
             if (comp != dotdot or (not initial_slashes and not new_comps) or
-                 (new_comps and new_comps[-1] == dotdot)):
+                (new_comps and new_comps[-1] == dotdot)):
```

Comment:
There's no need to change this line; we can reduce the diff here

```suggestion
                 (new_comps and new_comps[-1] == dotdot)):
```